file.go

```diff
@@ -578,9 +513,9 @@ func (f *File) Undo(isundo bool) (q0, q1 int, ok bool) {
 			f.mod = u.mod
 			f.treatasclean = false
 			f.b.Delete(u.p0, u.p0+u.n)
-			for _, text := range f.text {
-				text.deleted(u.p0, u.p0+u.n)
-			}
+			f.oeb.AllObservers(func(i interface{}) {
```

Comment:
This seems more complicated than necessary. Just have a function `deleted` in `ObservableEditableBuffer` that you call.